### ISSUE TITLE
String Comparison with an Null Object on one Site

### ISSUE THREAD

**Issue by stefanmeissner** [jtwig/jtwig#296]
The *RelationalOperations::eq* method has a null check implemented, but this will fail for string comparisons like this:
```
{% if recipient.attr != "" %}{{ recipient.attr }}{% else %}no or empty attribute{% endif %}
```
If *recipient.attr* is null, e.g. not set, this condition does not output the else branch. This is due to a missing String check for the other argument.

I will create a fix+test and a pull request.

What about comparisons of Date objects?

**Comment by thomas-p-wilson** [comment]
Could I bother you to try the same with my latest PR? It's possible I fixed
it there. I'd try it myself, but I'm on another project right now.

Date comparisons should work in my PR

**Comment by stefanmeissner** [comment]
Sure, I'll check this.

**Comment by stefanmeissner** [comment]
ATM I get stucked with this: *org.jtwig.exception.ParseException: Unknown tag: if*

But I have to change the parse call for the template, maybe I missed something there. I'll check this again later.

**Comment by thomas-p-wilson** [comment]
Ahh yes. The core extension, which contains pretty much everything except
the actual parsing stuff, is now optional. That way things can be
overridden, much like the way Twig does it. So right now you do a
JtwigConfigurationBuilder.newConfiguration() or something like that if I
recall. If you want the core stuff included, change it to
JtwigConfigurationBuilder.defaultConfiguration(). Pretty sure that's how it
goes. If that doesn't do the trick let me know and I'll look at the code.
Been a while since I touched it.

**Comment by stefanmeissner** [comment]
Close: *Wrong if syntax*

I'll try again those days.

**Comment by thomas-p-wilson** [comment]
Oh hah. Apparently I wasn't reading.

**Comment by stefanmeissner** [comment]
I don't get it. Here is my approach:
```
JtwigConfiguration configuration = JtwigConfigurationBuilder
				.defaultConfiguration();
Environment env = new Environment(configuration);

Template compiledTemplate = env.parse(new  FileResource(
				new File("./"), new File("test.twig")));
```

With test.twig = 
```
"{{ recipient.attr }}" 
```
everything is fine.
With test.twig = 
```
"{% if recipient.attr != "" %}{{ recipient.attr }}{% else %}no or empty attribute{% endif %}"
```
I get:
```
Exception in thread "main" org.jtwig.exception.ParseException: Wrong if syntax
Explanation: Input position (line 1, pos 7):
{% if recipient.attr != "" %}{{ recipient.attr }}{% else %}no or empty attribute{% endif %}
      ^


	at org.jtwig.extension.core.tokenparsers.IfStatementParser$$parboiled.rule(Unknown Source)
	at org.jtwig.parser.parboiled.JtwigContentParser$$parboiled.tokenParsers(Unknown Source)
	at org.jtwig.parser.parboiled.JtwigContentParser$$parboiled.content(Unknown Source)
	at org.jtwig.parser.parboiled.JtwigContentParser$$parboiled.start(Unknown Source)
	at org.jtwig.Environment.parse(Environment.java:93)
```

This is with your PR merged into the current master.

**Comment by thomas-p-wilson** [comment]
Sorry it's being a pain. I'll have a look. It's likely I screwed something
up.

**Comment by stefanmeissner** [comment]
There should be a solution where this is not true:

    {% if attr != '' %} 

if attr is null, e.g. not in the ModelMap.

I'll give it a try.

**Comment by stefanmeissner** [comment]
This issue should be resolved with Thomas PR #289

**Comment by lyncodev** [comment]
This is now working with the new version. Check the test https://github.com/jtwig/jtwig-core/blob/master/src/test/java/org/jtwig/integration/issues/Issue296Test.java .
